llvm/lib/Target/RISCV/RISCVInsertReadWriteCSR.cpp

```diff
@@ -55,6 +61,82 @@ char RISCVInsertReadWriteCSR::ID = 0;
 INITIALIZE_PASS(RISCVInsertReadWriteCSR, DEBUG_TYPE,
                 RISCV_INSERT_READ_WRITE_CSR_NAME, false, false)
 
+// TODO: Use more accurate rounding mode at the start of MBB.
+bool RISCVInsertReadWriteCSR::emitWriteRoundingModeOpt(MachineBasicBlock &MBB) {
+  bool Changed = false;
+  MachineInstr *LastFRMChanger = nullptr;
+  unsigned CurrentRM = RISCVFPRndMode::DYN;
+  std::optional<Register> SavedFRM;
```

Comment:
I don't think we need std::optional. We can use `SavedFRM.isValid()` to track validity. And we can use `SavedFRM = Register()` to make it invalid when we need to.